### ISSUE TITLE
Timestamp not passed.

### ISSUE THREAD

**Issue by jskilton** [logstash-plugins/logstash-output-loggly#12]
when using the loggly output plug-in, the @timestamp is not transmitted or used when sending the event to loggly.  If you have various inputs to logstash from nginx, apache , whether you use json logs or message flat text and grok parse them, if you match date and parse in the event so that logstash has @timestamp for the event, it does not make it to loggly. Instead the time teh post to loggly is used instead from the output plugin. i have found i needed to add a duplicate field named timestamp (performance and space costs) to the event if i expected loggly to be able to parse and use the correct timestamp instead of the time the http post came in.

**Comment by jgerry** [comment]
I just noticed this as well. It seems that Loggly wants a timestamp field called `timestamp` and not `@timestamp` - this is easily fixable with a mutate filter to add the needed field, or replace an existing `timestamp` field with the contents of `@timestamp` if you already have a a field called `timestamp`.

```
mutate {
  add_field => { "timestamp" => "%{@timestamp}" }
} 
```
or
```
mutate {
  replace => { "timestamp" => "%{@timestamp}" }
} 
```
I am also going to look at generating a pull request to fix the issue.

**Comment by webmat** [comment]
Resolved by #28, which was released as v5.0.0.
